### ISSUE TITLE
[Request] Support saving IPFS pages with their hash and periodically checking for newer "revisions"

### ISSUE THREAD

**Issue by dragon788** [wallabag/wallabag#3105]
IPFS essentially turns the web into a Git repository where the contents of a page is saved as a checksum, so you can tell if the page you are visiting has been modified since your last visit or altered from what the distributed web says the checksum should be (hacking/takeover/etc). This also allows you to more easily keep track of new updates to a page and cache them at a personal (or shared) level since the checksums ensure you are getting the page you really asked for.

https://ipfs.io/
https://github.com/ipfs/ipfs

**Comment by j0k3r** [comment]
This seems to be a really complex solution to implement. I mean, wallabag isn't designed to _periodically check_ for new version nor handle ipfs for the moment.
Do you have a _page_ that is in ipfs format? So we can check how we can integrate it.

Also, I think it won't be a priority anytime soon. 
Feel free to provide a PR about that

**Comment by thibaultmol** [comment]
You need to have IPFS desktop installed and the IPFS companion browser extension:
https://github.com/ipfs-shipyard/ipfs-desktop
https://github.com/ipfs-shipyard/ipfs-companion

IPFS's own blog is available as a regular site, but when visited with ipfs installed it goes trough IPFS.

(IFPS websites can be accessed without needing to have ipfs installed, then they go trough an IPFS Gateway which basically does that for you. Cloudflare has a free one and so does IPFS themselves:
https://cloudflare-ipfs.com/ipfs/INSERT-THE-IPFS-HASH?filename=IMG_20200727_172605.jpg
https://ipfs.io/ipfs/INSERT-THE-IPFS-HASH

And as I showed in the cloudflare link above, it can also have a suffix like that for the filename (although that is optional)

This video explains ipfs pretty well: https://www.youtube.com/watch?v=5Uj6uR3fp-U
So basically everything is linked to hash. And they can also be updated by using the metadata (so a versioning system pretty much)
